sky/templates/jobs-controller.yaml.j2

```
@@ -26,6 +26,10 @@ setup: |
   echo 'export SKYPILOT_DEV=1' >> ~/.bashrc
   {% endif %}
 
+  {% if controller_envs.get("SKYPILOT_DEV") %}
+  type sky-env > /dev/null 2>&1 || echo 'alias sky-env="{{ sky_activate_python_env }}"' >> ~/.bashrc
+  {% endif %}
```

Can we reuse the `is_dev` conditional check above and wrap it into the same block?
```suggestion
  echo 'export SKYPILOT_DEV=1' >> ~/.bashrc
  type sky-env > /dev/null 2>&1 || echo 'alias sky-env="{{ sky_activate_python_env }}"' >> ~/.bashrc
  {% endif %}
```

---

Also, IIUC `type` checks symbols in the active shell. While it will work here, do you think something like this would be a more robust check:
```
  if ! grep -q "alias sky-env=" ~/.bashrc; then
    echo 'alias sky-env="{{ sky_activate_python_env }}"' >> ~/.bashrc
  fi
```

---

+1 to the grep solution, especially since I don't think it's guaranteed that the commands here are running in bash (could be some other sh like dash)

---

Both suggestions make sense. Let me try again!

---

Wow, yeah, that's apparently been broken for more than a year. #2458 removed `is_dev` but apparently didn't remove this use. Good catch!
Could you move the `export SKYPILOT_DEV=1` into your new conditional, delete the old one, and add a similar grep check? e.g.
```
grep -q "export SKYPILOT_DEV=" ~/.bashrc || echo 'export SKYPILOT_DEV=1' >> ~/.bashrc
```

---

Sorry for the wait (work taking up a bit of my time). I've made the changes and ran a quick sanity check - seems to be good.

I am realizing it would be good to refactor this to a list of aliases outside of the jinja template, but I'll leave that for a future PR.